test/system/src/main/java/io/pravega/test/system/framework/services/docker/BookkeeperDockerService.java

```diff
@@ -81,7 +82,7 @@ private ServiceSpec setServiceSpec() {
         stringList.add(env4);
 
         final TaskSpec taskSpec = TaskSpec
-                .builder().restartPolicy(RestartPolicy.builder().maxAttempts(0).condition("none").build())
+                .builder().restartPolicy(RestartPolicy.builder().maxAttempts(1).condition(RESTART_POLICY_ANY).build())
```

Comment:
I have a couple of clarification questions.
- Could provide more information about the "any" restart policy v/s "on_failure"?
- Does setting maxAttempts imply it is restarted only on the first failure?
- Currently in the K8s based test framework a bookie / or any other pravega container restarts would lead to a test failure. Would the behavior be the same here ?

Reply:
"on-failure" will restart the container only if it exited with error."any" will restart the container , even if the container exited with no errors. In our bookie tests, one of the bookie was got into running state and others were shutting down without any errors.

Max attempts will restart the container only on the first failure.

I am not sure whether the restart of container cause test failure.

Reply:
In a conversation with @adrianmo, it looks like this change is necessary due to the following reason:
1) With the new version of Bookkeeper, if all the bookies are started at once they will try to acquire the Zookeeper lock to create the necessary metadata.
2) Only one of the Bookies will succeed on doing so, and the rest of Bookies just terminate instead of waiting until the metadata is initialized. 
3) The expectation for the rest of Bookies is that once they are restarted, the metadata will be initialized by the Bookie that acquired the lock in the first place.
Is this correct @anishakj? Does this clarifies the motivation for this change @shrids?

Reply:
yes it does. Thanks @RaulGracia @anishakj